lib/platform/linux-i2c.c

```diff
@@ -162,41 +202,191 @@ static gasptr_t i2c_gas_map(struct switchtec_dev *dev, int writeable,
 	return dev->gas_map;
 }
 
-/* One I2C transaction can write a maximum of 28 bytes */
-#define I2C_MAX_WRITE 28
+static uint8_t i2c_gas_cap_get(struct switchtec_dev *dev)
+{
+	int ret;
+	struct switchtec_i2c *idev = to_switchtec_i2c(dev);
 
-static void i2c_gas_write(struct switchtec_dev *dev, void __gas *dest,
-			  const void *src, size_t n)
+	struct i2c_msg msgs[2];
+	struct i2c_rdwr_ioctl_data rwdata = {
+		.msgs = msgs,
+		.nmsgs = 2,
+	};
+
+	uint8_t command_code = CMD_GET_CAP;
+	uint8_t rx_buf[2];
+
+	uint8_t msg_0_pec, pec;
+	uint8_t retry_count = 0;
+
+	msgs[0].addr = msgs[1].addr = idev->i2c_addr;
+	msgs[0].flags = 0;
+	msgs[0].len = 1;
+	msgs[0].buf = &command_code;
+
+	msgs[1].flags = I2C_M_RD;
+	msgs[1].len = 2;
+	msgs[1].buf = rx_buf;
+
+	do {
+		ret = ioctl(idev->fd, I2C_RDWR, &rwdata);
+		if (ret < 0) {
+			/*
+			 * For now, this is all we can do. If it's a problem we'll
+			 * have to refactor all GAS access functions to return an
+			 * error code.
+			 */
+			perror("Could not communicate with I2C device");
+			exit(1);
```

Comment:
Don't call perror from the library. Return an error number so the calling code can call perror.